Code/Framework/AzFramework/Platform/Common/Xcb/AzFramework/XcbNativeWindow.cpp

```diff
@@ -219,17 +243,127 @@ namespace AzFramework
         }
     }
 
+    void XcbNativeWindow::DrawSplash()
+    {
+        auto settingsRegistry = AZ::SettingsRegistry::Get();
+        AZ::SettingsRegistryInterface::FixedValueString splashScreenImagePath;
+        AZ::SettingsRegistryInterface::FixedValueString assetCachePath;
+        static const AZStd::string SplashLogoSetregPath = "/O3DE/xcb/SplashScreenImagePath";
+        static const AZStd::string AssetsSetregPath = "/O3DE/Runtime/FilePaths/CacheProjectRootFolder";
+
+        if (!(settingsRegistry && settingsRegistry->Get(splashScreenImagePath, SplashLogoSetregPath.c_str())))
+        {
+            printf("SplashScreenImagePath not found\n");
+            return;
+        }
+        if (!(settingsRegistry && settingsRegistry->Get(assetCachePath, AssetsSetregPath.c_str())))
+        {
+            printf("Failed to grab cache folder\n");
+            return;
+        }
+        const AZStd::string splashScreenImagePath2 =
+            AZStd::string::format("%s/linux/%s", assetCachePath.c_str(), splashScreenImagePath.c_str());
+
+        FILE* fp = fopen(splashScreenImagePath2.c_str(), "rb");
+        if (!fp)
+        {
+            printf("Failed to open image %s \n", splashScreenImagePath2.c_str());
+            return;
+        }
+        // Initialize libpng
+        png_structp png_ptr = png_create_read_struct(PNG_LIBPNG_VER_STRING, NULL, NULL, NULL);
+        png_infop info_ptr = png_create_info_struct(png_ptr);
+        png_init_io(png_ptr, fp);
+        png_read_info(png_ptr, info_ptr);
+
+        // Read the PNG into an in-memory format that XCB can use
+        int width = png_get_image_width(png_ptr, info_ptr);
+        int height = png_get_image_height(png_ptr, info_ptr);
+
+        png_bytep row_pointers[height];
+        for (int y = 0; y < height; y++)
+        {
+            row_pointers[y] = (png_byte*)malloc(png_get_rowbytes(png_ptr, info_ptr));
+        }
+        png_read_image(png_ptr, row_pointers);
+        fclose(fp);
+        unsigned int stride = width * 4;
+        uint8_t* data = (uint8_t*)malloc(height * stride);
```

Comment:
Use smth from C++ not C.

Reply:
Whole PNG code should be contained in a separated method/function.

Reply:
As explained this malloc is needed as xcb expects ability to free data, variable use of vector results in heap corruption.